### PR TITLE
Merge 2.1 into 2.2

### PR DESCRIPTION
**Pull request by jkotalik**
Porting infrastructure changes forward.

### CODE REVIEW

File: build/artifacts.props [hunk 1]
```diff
@@ -182,6 +182,7 @@
     <PackageArtifact Include="Microsoft.EntityFrameworkCore" Category="ship" />
     <PackageArtifact Include="Microsoft.Extensions.ApiDescription.Design" Category="noship" />
     <PackageArtifact Include="Microsoft.Extensions.ApplicationModelDetection" Category="noship" />
+    <PackageArtifact Include="Microsoft.Extensions.Buffers.Testing.Sources" Category="noship"/>
```
Comment (natemcmaster) [inline build/artifacts.props]: Can you also add back the config for Microsoft.Extensions.Buffers.MemoryPool.Sources? This was a new .sources package in 2.2 (found in https://github.com/aspnet/AspNetCore/tree/release/2.2/src/Shared/Buffers.MemoryPool)

File: eng/Baseline.xml [hunk 2]
```diff
@@ -45,6 +45,7 @@
   <Package Id="Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv" Version="2.2.0" />
   <Package Id="Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets" Version="2.2.0" />
   <Package Id="Microsoft.AspNetCore.Server.Kestrel" Version="2.2.0" />
+  <Package Id="Microsoft.AspNetCore.Server.HttpSys" Version="2.2.0" />
```
Comment (Tratcher) [inline eng/Baseline.xml]: Is this supposed to be alphabetical?

File: eng/Dependencies.props [hunk 3]
```diff
@@ -48,6 +48,7 @@
     <LatestPackageReference Include="System.Data.SqlClient" Version="$(SystemDataSqlClientPackageVersion)" />
     <LatestPackageReference Include="System.Memory" Version="$(SystemMemoryPackageVersion)" />
     <LatestPackageReference Include="System.Net.WebSockets.WebSocketProtocol" Version="$(SystemNetWebSocketsWebSocketProtocolPackageVersion)" />
+    <LatestPackageReference Include="System.Net.Http.WinHttpHandler" Version="$(SystemNetHttpWinHttpHandlerPackageVersion)" />
```
Comment (Tratcher) [inline eng/Dependencies.props]: Sort?

**Comment by jkotalik** [comment]
🆙 📅

**Comment by natemcmaster** [comment]
Still getting this build failure:
```
2018-12-06T01:42:17.4588937Z D:\a\1\s\build\repo.targets(321,5): error : Expected the following packages, but they were not found: - Internal.WebHostBuilderFactory.Sources [D:\a\.dotnet\buildtools\korebuild\2.2.1-build-20181203.2\KoreBuild.proj]
2018-12-06T01:42:17.4589739Z D:\a\1\s\build\repo.targets(321,5): error :  - Microsoft.Extensions.Buffers.MemoryPool.Sources [D:\a\.dotnet\buildtools\korebuild\2.2.1-build-20181203.2\KoreBuild.proj]
2018-12-06T01:42:17.4590144Z D:\a\1\s\build\repo.targets(321,5): error :  - Microsoft.Extensions.Buffers.Testing.Sources [D:\a\.dotnet\buildtools\korebuild\2.2.1-build-20181203.2\KoreBuild.proj]
2018-12-06T01:42:17.4590390Z D:\a\1\s\build\repo.targets(321,5): error :  [D:\a\.dotnet\buildtools\korebuild\2.2.1-build-20181203.2\KoreBuild.proj]
```

**Pull request closed without merging**